piper/window.py

```diff
@@ -183,3 +173,16 @@ def _on_device_selected(self, perspective, device):
 
     def _get_child(self, name):
         return self.stack_perspectives.get_child_by_name(name)
+
+    def _perspective_add_back_button(self, perspective):
+        button_back = Gtk.Button.new_from_icon_name("go-previous-symbolic",
+                                                    Gtk.IconSize.BUTTON)
+        button_back.set_visible(len(self.ratbag.devices) > 1)
+        button_back.connect("clicked", lambda button, ratbag:
+                            self._present_welcome_perspective(ratbag.devices),
+                            self.ratbag)
+        self.ratbag.connect("notify::devices", lambda ratbag, pspec:
+                            button_back.set_visible(len(ratbag.devices) > 1))
```

Comment:
there is no `self.ratbag`, we only connect the signals to it and then drop it. You'll have to pass this down as argument to the function. And please test it, I can't see how this would've worked, maybe you forgot to commit some local change?


Coincidentally, can you in the process fix the indentation so it reads
```
button_back.connect("clicked", 
                    lambda button, ratbag: self._present....,
                    ratbag)
```
Because right now it looks like the `self._present` line is an argument to `connect()`

Reply:
> And please test it, I can't see how this would've worked, maybe you forgot to commit some local change?

Sorry, I tested the back button by making it visible manually with GTK's inspector, instead of spawning the ratbagd test device (i.e., I was being lazy).

I fixed this and the indentation.